app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/CaptureLocationViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package org.greenstand.android.TreeTracker.viewmodels
+
+import android.location.Location
+import android.util.Base64
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.work.WorkManager
+import com.google.gson.Gson
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.greenstand.android.TreeTracker.database.TreeTrackerDAO
+import org.greenstand.android.TreeTracker.database.entity.LocationCaptureEntity
+import org.greenstand.android.TreeTracker.managers.UserLocationManager
+import org.greenstand.android.TreeTracker.managers.UserManager
+import timber.log.Timber
+import kotlin.coroutines.CoroutineContext
+
+class CaptureLocationViewModel(

```

Comment:
It seems like we're using this viewmodel less like a viewmodel, and more like a model layer objected with a limited scope.
The two views that use this ViewModel already have Viewmodels, which can get confusing. Can we look into using Koins scoping feature? Perhaps we can create a tree capture scope to help limit the capture time of location.